components/services/brave_wallet/zcash/zcash_decoder.cc

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/services/brave_wallet/zcash/zcash_decoder.h"
+
+#include <utility>
+
+#include "base/big_endian.h"
+#include "brave/components/brave_wallet/common/proto/protobuf_utils.h"
+#include "brave/components/brave_wallet/common/proto/zcash_grpc_data.pb.h"
+
+namespace brave_wallet {
+
+ZCashDecoder::ZCashDecoder() = default;
+
+ZCashDecoder::~ZCashDecoder() = default;
+
+void ZCashDecoder::ParseRawTransaction(const std::string& data,
+                                       ParseRawTransactionCallback callback) {
+  zcash::RawTransaction result;
+  auto serialized_message = ResolveSerializedMessage(data);
+  if (!serialized_message ||
+      !result.ParseFromString(serialized_message.value())) {
+    std::move(callback).Run(nullptr, "Failed to parse message");
```

Comment:
Let's move this string `"Failed to parse message"` to a constant or even an anymouns function. So further l11n for zcash will touch less code

Reply:
Or do we even need that same error meessage? Maybe just return a single optional. Most likely all kinds of decoding failures would be equal for caller.

Reply:
Done